src/digital_identity_service/index.js

```diff
@@ -1,16 +1,103 @@
+'use strict';
+
+const config = require('../../config');
 const Validation = require('../yoti_common/validation');
 
+const { RequestBuilder } = require('../request/request.builder');
+const { Payload } = require('../request/payload');
+const ShareSessionCreateResult = require('./share.session.create.result');
+
+const PolicyBuilder = require('./policy/policy.builder');
+const WantedAttributeBuilder = require('./policy/wanted.attribute.builder');
+const WantedAnchorBuilder = require('./policy/wanted.anchor.builder');
+const ConstraintsBuilder = require('./policy/constraints.builder');
+const SourceConstraintBuilder = require('./policy/source.constraint.builder');
+const ExtensionBuilder = require('./extension/extension.builder');
+const LocationConstraintExtensionBuilder = require('./extension/location.constraint.extension.builder');
+const TransactionalFlowExtensionBuilder = require('./extension/transactional.flow.extension.builder');
+const ThirdPartyAttributeExtensionBuilder = require('./extension/third.party.attribute.extension.builder');
+const ShareSessionBuilder = require('./share.session.builder');
+const ShareNotificationBuilder = require('./share.notification.builder');
+const ShareSession = require('./share.session');
+
+const DEFAULT_API_URL = config.yoti.digitalIdentityApi;
+
 class DigitalIdentityService {
-  constructor(sdkId, pem, { apiUrl }) {
+  constructor(sdkId, pem, { apiUrl = DEFAULT_API_URL } = {}) {
     Validation.isString(sdkId, 'sdkId');
     Validation.notNullOrEmpty(pem, 'pem');
 
     this.sdkId = sdkId;
     this.pem = pem;
     this.apiUrl = apiUrl;
   }
+
+  /**
+   *
+   * @param shareSessionConfig
+   * @returns {Promise<ShareSessionCreateResult>}
+   */
+  async createSession(shareSessionConfig) {
+    Validation.instanceOf(shareSessionConfig, ShareSession, 'shareSessionConfig');
+
+    const payload = new Payload(shareSessionConfig);
+
+    const requestBuilder = new RequestBuilder()
+      .withBaseUrl(this.apiUrl)
+      .withHeader('X-Yoti-Auth-Id', this.sdkId)
+      .withPemString(this.pem)
+      .withEndpoint('/v2/sessions')
+      .withQueryParam('appId', this.sdkId)
+      .withMethod('POST')
+      .withPayload(payload);
+
+    const request = requestBuilder.build();
+
+    try {
+      const response = await request.execute().catch((error) => {
```

Comment:
Is it correct to still have the `catch` here? Isn't the `try`/`catch` going to handle this now, since there is an `await`?

Reply:
This chained `catch` is to handle the `execute` error, but the next `catch` is for handling the error of creating an instance of `ShareSessionCreateResult`. 

So I had to chain `catch` with `execute` or have nested `try` `catch`.

What do you think, should we do it this way or have a nested `try` `catch`?

Reply:
Well, if the chained `catch` is handling the error from `execute`, why is it in the `try` block?

Then, I'm not sure how the chained `catch` works synchronously. What happens if `execute` errors, and goes to into the chained `catch`? Will it throw that error in the main thread?

Reply:
I think Majeed tried to respect the previous implementation, which is handling the call first, then the parsing of the data.
Yet, it is not quite equivalent to what existed (comparing with DynamicScenario service).

The .catch() will print the error like before and then throw the error, yet, this one is now caught by the try/catch, which will print the error again (yet with a different message), and finally throw it.

If we just want the same, we can simply use only try/catch, without nesting though.

```js
try {
  // do the call and assign result to variable
} catch(e) {
  // print error with associated message
 throw e
}

try {
  // convert result variable to final result class
} catch(e) {
  // print error with associated message
 throw e
}
```